tutorindigo/templates/indigo/lms/static/js/dark-theme.js

```diff
@@ -1,8 +1,38 @@
 $(document).ready(function() {
     'use strict';
 
-    const enableDarkTheme = {% if INDIGO_ENABLE_DARK_THEME %}true{% else %}false{% endif %};
-    if (enableDarkTheme){
-        $('body').addClass("indigo-dark-theme");
+    const themeCookie = 'indigo-toggle-dark';
+
+    function applyThemeOnPage(){
+      const theme = $.cookie(themeCookie);
+      {% if INDIGO_ENABLE_DARK_TOGGLE %}
+      $('body').toggleClass("indigo-dark-theme", theme === 'dark');       // append or remove dark-class based on cookie-value
+      // update expiry
+      $.cookie(themeCookie, theme, { domain: window.location.hostname, expires: 90, path: '/' });
```

Comment:
how is this different from operation in above file?

Reply:
The `applyThemeOnPage()` function is the minimum criterion directly affecting whether the dark theme is applied. Instead of adding to the whole file, we focus specifically on this function. 
But there are no significant performance gains in either way because LMS has to compile JS file whenever building an image.